docs/design/analytics.md

```diff
@@ -0,0 +1,114 @@
+
+**DO NOT ADD `x-user-analytics` FOR VALUES THAT ARE USER IDENTIFIABLE (PII), FOR EXAMPLE A
+PROJECT NAME TO BUILD OR A MODULE NAME.**
+
+Note: There's a limit of 20 custom dimensions.
+
+### List Of All Dimensions
+<!--DIMENSIONS_TABLE_BEGIN-->
+| Id | Category | Flag | Type | File / Description |
+|:---:|:---|:---|:---|---:|
+| 1 | `generate`, `new` | `--dryRun`      | `Boolean` | |
+| 2 | `generate`, `new` | `--force`       | `Boolean` | |
+| 3 | `generate`, `new` | `--interactive` | `Boolean` | |
+| 4 | `generate` | `--skipInstall` | `Boolean` | |
+| 5 | `generate` | `--style` | `String` | |
+| 6 | `add` | `--collection` | `String` | Only for packages that we control (see safelist in `add-impl.ts`). |
+| 7 | `build`, `serve` | `--buildEventLog` | `Boolean` | If the flag was used (does not report its value). See `build-impl.ts`. |
+| 8 | `generate`, `new` | `--enableIvy` | `Boolean` | |
+| 9 | `generate` | `--inlineStyle` | `Boolean` | |
+| 10 | `generate` | `--inlineTemplate` | `Boolean` | |
+| 11 | `generate` | `--viewEncapsulation` | `String` | |
+| 12 | `generate` | `--skipTests` | `Boolean` | |
+| 13 | `build` | `--aot` | `Boolean` | |
+| 14 | `generate` | `--minimal` | `Boolean` | |
+| 15 | `generate` | `--lintFix` | `Boolean` | |
+| 16 | `build` | `--optimization` | `Boolean` | |
+| 17 | `generate` | `--routing` | `Boolean` | |
+| 18 | `generate` | `--skipImport` | `Boolean` | |
+| 19 | `generate` | `--export` | `Boolean` | |
+| 20 | `generate` | `--entryComponent` | `Boolean` | |
+<!--DIMENSIONS_TABLE_END-->
+
+# Metrics
+
+### List of All Metrics
+<!--METRICS_TABLE_BEGIN-->
+| Id  | Type | Description |
+|:---:|:-----|-------------|
+| 1 | `Number` | CPU count |
+| 2 | `Number` | Average CPU speed |
+| 3 | `Number` | RAM Count |
+| 4 | `Number` | Node Version (Major.Minor) |
+<!--METRICS_TABLE_END-->
+
+# Operating System and Node Version
+A User Agent string is built to "fool" Google Analytics into reading the Operating System and
+version fields from it. The base dimensions are used for those.
+
+# Debugging
+Using `DEBUG=universal-analytics` will report all calls to the universal-analytics library,
+including queuing events and sending them to the server.
+
+Using `DEBUG=ng:analytics` will report additional information regarding initialization and
+decisions made during the usage analytics process, e.g. if the user has analytics disabled.
+
+Using `DEBUG=ng:analytics:command` will show the decisions made by the command runner.
+
+Using `DEBUG=ng:analytics:log` will show what we actually send to GA.
+
+See [the `debug` NPM library](https://www.npmjs.com/package/debug) for more information.
+
+# Disabling Telemetry
+There are 2 ways of disabling usage analytics:
+
+1. using `ng analytics off` (or changing the global configuration file yourself). This is the same
+   as answering "No" to the prompt.
+1. There is an `NG_CLI_TELEMETRY` environment variable that overrides the global configuration.
```

Comment:
Cannot find any code referencing `NG_CLI_TELEMETRY`. Am I missing something?

Reply:
It's `NG_CLI_ANALYTICS` now. Renamed in the md.